src/widgets/panel/colorgrabwidget.cpp

```diff
@@ -0,0 +1,184 @@
+#include "colorgrabwidget.h"
+#include "sidepanelwidget.h"
+
+#include "colorutils.h"
+#include "confighandler.h"
+#include "overlaymessage.h"
+#include "src/core/qguiappcurrentscreen.h"
+#include <QApplication>
+#include <QDebug>
+#include <QKeyEvent>
+#include <QPainter>
+#include <QScreen>
+#include <QShortcut>
+#include <QTimer>
+#include <stdexcept>
+
+// Width (= height) and zoom level of the widget before the user clicks
+#define WIDTH1 80
+#define ZOOM1 8
+// Width (= height) and zoom level of the widget after the user clicks
+#define WIDTH2 165
+#define ZOOM2 15
+
+// NOTE: WIDTH1(2) should be divisible by ZOOM1(2) for best precision.
```

Comment:
@mmahmoudian Btw, you can tweak these if you feel so inclined.

Reply:
I think these are good enough. I don't have a 4k display though, so I don't know how small they look in such monitor.